packages/account/src/Components/poi/idv-document-submit/idv-document-submit.tsx

```diff
@@ -18,220 +17,233 @@ import BackButtonIcon from 'Assets/ic-poi-back-btn.svg';
 import IDVForm from 'Components/forms/idv-form';
 import PersonalDetailsForm from 'Components/forms/personal-details-form';
 import FormSubHeader from 'Components/form-sub-header';
+import { observer, useStore } from '@deriv/stores';
+import FormikConfirmationCheckbox from 'Components/formik-confirmation-checkbox';
+
+type TIdvDocumentSubmit = {
+    handleBack?: () => void;
+    handleViewComplete?: () => void;
+    is_from_external?: boolean;
+    selected_country?: object;
+};
 
-const IdvDocumentSubmit = ({
-    handleBack,
-    handleViewComplete,
-    selected_country,
-    is_from_external,
-    account_settings,
-    getChangeableFields,
-}) => {
-    const visible_settings = ['first_name', 'last_name', 'date_of_birth'];
-    const form_initial_values = filterObjProperties(account_settings, visible_settings) || {};
-
-    if (form_initial_values.date_of_birth) {
-        form_initial_values.date_of_birth = toMoment(form_initial_values.date_of_birth).format('YYYY-MM-DD');
-    }
-
-    const changeable_fields = [...getChangeableFields()];
-
-    const initial_values = {
-        document_type: {
-            id: '',
-            text: '',
-            value: '',
-            example_format: '',
-            sample_image: '',
-        },
-        document_number: '',
-        ...form_initial_values,
-    };
-
-    const getExampleFormat = example_format => {
-        return example_format ? localize('Example: ') + example_format : '';
-    };
-
-    const shouldHideHelperImage = document_id => document_id === IDV_NOT_APPLICABLE_OPTION.id;
-
-    const isDocumentTypeValid = document_type => {
-        if (!document_type?.text) {
-            return localize('Please select a document type.');
+const IdvDocumentSubmit = observer(
+    ({ handleBack, handleViewComplete, selected_country, is_from_external }: TIdvDocumentSubmit) => {
+        const {
+            client: { account_settings, getChangeableFields },
+        } = useStore();
```

Comment:
```suggestion
        const { client: } = useStore();
        const { account_settings, getChangeableFields } = client;
```

can we please use this approach as we do though whole app, as it is more readable

Reply:
Yes, and the multiline makes more sense in readability when there's a lot going on in terms of destructuring. Right now we only have one line as I would keep this as is. It's the same as destructuring in a new line, better do it in the same place

Reply:
well, actually the suggested code fits into two lines instead of three :)
IMHO, it looks like more readable :)
you can ask for other options on this :rocket:

Reply:
I feel when u r going to just use 2 or max 3 props from a object, u can just pick those props from the object. If there are more than 3 then u can destructure in a new line. This is a **nit pick** :)